plugins/plugin-languageserver/che-plugin-languageserver-ide/src/main/java/org/eclipse/che/plugin/languageserver/ide/editor/quickassist/ApplyWorkspaceEditAction.java

```diff
@@ -147,6 +188,141 @@ public void applyWorkspaceEdit(WorkspaceEdit edit) {
             });
   }
 
+  private Promise<Void> applyResourceChanges(Notification notification, WorkspaceEdit edit) {
+    List<Either<ResourceChange, TextDocumentEdit>> resourceChanges = edit.getResourceChanges();
+    if (resourceChanges.isEmpty()) {
+      return promises.resolve(null);
+    }
+
+    ArrayOf<Promise<?>> changesPromises = elemental.util.Collections.arrayOf();
+
+    for (Either<ResourceChange, TextDocumentEdit> rChange : resourceChanges) {
+      if (rChange.isRight()) {
+        continue;
+      }
+
+      ResourceChange change = rChange.getLeft();
+      if (change == null) {
+        continue;
+      }
+      String newUri = change.getNewUri();
+      String current = change.getCurrent();
+
+      Path path = toPath(newUri).removeFirstSegments(1).makeAbsolute();
```

Comment:
Transformation of file uri's to paths is usually done in WSMaster, so results from lsp servers contain either workspace paths of non-file URI's. See LanguageServiceUtils.removePrefixURI.

Reply:
GWT doesn't support java.net.* so we can't use `LanguageServiceUtils.java` on the client. I've created utility class `org.eclipse.che.api.languageserver.util.URIUtil` which transforms uri to path.

Reply:
Not what I mean, I've tried to explain in chat.

Reply:
fixed